lib/routes/veeam/put.js

```diff
@@ -0,0 +1,83 @@
+const async = require('async');
+const { parseString } = require('xml2js');
+const { receiveData, isSystemXML, getFileToBuild } = require('./utils');
+const { s3routes, errors } = require('arsenal');
+const metadata = require('../../metadata/wrapper');
+const parseSystemSchema = require('./schemas/system');
+const parseCapacitySchema = require('./schemas/capacity');
+const writeContinue = require('../../utilities/writeContinue');
+
+const { responseNoBody, responseXMLBody } = s3routes.routesUtils;
+
+/**
+ * Puts a veeam capacity or system file in the bucket metadata.
+ * Logic ensures consistency of the data and metadata.
+ *
+ * @param {object} request - request object
+ * @param {object} response - response object
+ * @param {object} bucketMd - bucket metadata from the db
+ * @param {object} log - logger object
+ * @returns {undefined} -
+ */
+function putVeeamFile(request, response, bucketMd, log) {
+    if (!bucketMd) {
+        return errors.NoSuchBucket;
+    }
+
+    return async.waterfall([
+        next => {
+            // Extract the data from the request, keep it in memory
+            writeContinue(request, response);
+            return receiveData(request, log, next);
+        },
+        (value, next) => parseString(value, { explicitArray: false }, (err, parsed) => {
+            // Convert the received XML to a JS object
+            if (err) {
+                return next(errors.MalformedXML);
+            }
+            return next(null, parsed);
+        }),
+        (parsedXML, next) => {
+            const capabilities = bucketMd._capabilities || {
+                VeeamSOSApi: {},
+            };
+            // Initialize Veeam capability if not already there
+            if (!capabilities.VeeamSOSApi) {
+                capabilities.VeeamSOSApi = {};
+            }
```

Comment:
not really needed:
* the 'default' initi is redundant with the later 'Initialize Veeam capability if not already there'
* all these are actually not needed, since we set it anyway line 62: ` capabilities.VeeamSOSApi = { ...`

i guess we could simply do this (at line 62):
```
            capabilities.VeeamSOSApi = {
                ...(capabilities.VeeamSOSApi || {}),
                ...file.value,
            };
```